device/tt_device/tt_device.cpp

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "device/tt_device/tt_device.h"
+
+#include "device/tt_device/blackhole_tt_device.h"
+#include "device/tt_device/grayskull_tt_device.h"
+#include "device/tt_device/wormhole_tt_device.h"
+
+namespace tt::umd {
+
+std::unique_ptr<TTDevice> TTDevice::create(architecture architecture) {
+    switch (architecture) {
+        case architecture::blackhole: return std::make_unique<BlackholeTTDevice>();
+        case architecture::grayskull: return std::make_unique<GrayskullTTDevice>();
+        case architecture::wormhole:
```

Comment:
not sure whether we should still support wormhole (without b0), I think we can deprecate that arch

Reply:
I don't want to make the decision to drop this for now. It's not a big cost to support it. When we choose to drop it, we will easily remove it.